nlmod/dims/resample.py

```diff
@@ -92,6 +92,30 @@ def get_xy_mid_structured(extent, delr, delc, descending_y=True):
         raise TypeError("unexpected type for delr and/or delc")
 
 
+def get_delr(ds):
+    """Get the distance along rows (delr) fromythe x-coordinate of a model dataset"""
+    assert ds.gridtype == "structured"
+    x = (ds.x - ds.extent[0]).values
+    delr = _get_delr_from_x(x)
+    return delr
+
+
+def get_delc(ds):
+    """Get the distance along columns (delr) from the x-coordinate of a model dataset"""
+    assert ds.gridtype == "structured"
+    y = (ds.extent[3] - ds.y).values
+    delc = _get_delr_from_x(y)
```

Comment:
Kind of counterintuitive to call _get_delr_from_x to get_delc_from_y. Maybe rename the private method to _get_delta_along_axis or something similar?